orchagent/portsorch.cpp

```diff
@@ -3647,6 +3693,88 @@ void PortsOrch::doPortTask(Consumer &consumer)
+                             "Set port %s link event damping algorithm to %s",
+                             p.m_alias.c_str(), m_portHlpr.getDampingAlgoStr(pCfg).c_str()
+                         );
+                    }
+                }
+
+                sai_redis_link_event_damping_algo_aied_config_t aied_config = {
+                    p.m_max_suppress_time,
+                    p.m_suppress_threshold,
+                    p.m_reuse_threshold,
+                    p.m_decay_half_life,
+                    p.m_flap_penalty,
+                };
+
+                if (pCfg.link_event_damping.max_suppress_time.is_set)
+                {
+                    aied_config.max_suppress_time = pCfg.link_event_damping.max_suppress_time.value;
+                }
+                if (pCfg.link_event_damping.decay_half_life.is_set)
+                {
+                    aied_config.decay_half_life = pCfg.link_event_damping.decay_half_life.value;
+                }
+                if (pCfg.link_event_damping.suppress_threshold.is_set)
+                {
+                    aied_config.suppress_threshold = pCfg.link_event_damping.suppress_threshold.value;
+                }
+                if (pCfg.link_event_damping.reuse_threshold.is_set)
+                {
+                    aied_config.reuse_threshold = pCfg.link_event_damping.reuse_threshold.value;
+                }
+                if (pCfg.link_event_damping.flap_penalty.is_set)
+                {
+                    aied_config.flap_penalty = pCfg.link_event_damping.flap_penalty.value;
+                }
+
+                bool config_changed = !(aied_config.max_suppress_time == p.m_max_suppress_time &&
+                                        aied_config.decay_half_life == p.m_decay_half_life &&
+                                        aied_config.suppress_threshold == p.m_suppress_threshold &&
+                                        aied_config.reuse_threshold == p.m_reuse_threshold &&
+                                        aied_config.flap_penalty == p.m_flap_penalty);
+
+                if (config_changed) {
+                    if (!setPortLinkEventDampingAiedConfig(p, pCfg.link_event_damping_config.value))
+                    {
+                        SWSS_LOG_ERROR("Failed to set port %s link event damping config", p.m_alias.c_str());
+                        it = taskMap.erase(it);
+                        continue;
+                    }
+
+                    p.m_max_suppress_time = aied_config.max_suppress_time;
+                    p.m_decay_half_life = aied_config.decay_half_life;
+                    p.m_suppress_threshold = aied_config.suppress_threshold;
+                    p.m_reuse_threshold = aied_config.reuse_threshold;
+                    p.m_flap_penalty = aied_config.flap_penalty;
+                    m_portList[p.m_alias] = p;
+
+                    SWSS_LOG_NOTICE("Set port %s link event damping config successfully", p.m_alias.c_str());
```

Comment:
there is already a log in setPortLinkEventDampingAiedConfig, how about changing that to NOTICE and remove all logs here?

Reply:
I removed some log messages that were also in setPortLinkEventDampingAiedConfig.